### PR TITLE
Fix 500 errors caused by templates

### PR DESCRIPTION
**Pull request by zelima**
- includes small helper function that validates descriptor - tested
- logic updated and test included to check invalid licenses 

refs #355

### CODE REVIEW

File: app/logic/__init__.py [hunk 1]
```diff
@@ -218,3 +218,26 @@ def generate_signed_url():
                             props=filedata[relative_path])
         res_payload['filedata'][relative_path] = response.build_file_information()
     return res_payload
+
+
+#### helpers
+
+def validate_for_template(descriptor):
+    '''
+    Cheks if descriptor fields have expected type for template Eg:
```
Comment (anuveyatsu) [inline app/logic/__init__.py]: small typo: "Checks ..."

Comment (anuveyatsu) [inline app/logic/__init__.py]: Probably it's better to rewrite:
"This function validates field types in the descriptor, e.g.,  `license` property should be a list."

File: tests/logic/test_logic.py [hunk 2]
```diff
@@ -175,3 +175,34 @@ def tearDown(self):
...
+
+    def test_validate_for_jinja_returns_descriptor_if_licenses_is_list(self):
+        self.descriptor['licenses'] = []
+        descriptor = validate_for_template(self.descriptor)
+        self.assertEqual(self.descriptor, descriptor)
+
+    def test_validate_for_jinja_modifies_descriptor_if_licenses_is_dict(self):
+        self.descriptor['licenses'] = {'url': 'test/url', 'type': 'Test'}
+        descriptor = validate_for_template(self.descriptor)
+        self.assertEqual(descriptor['license'], 'Test')
```
Comment (anuveyatsu) [inline tests/logic/test_logic.py]: so `licenses` (plural) then it is a list, and if `license` singular then it is a string?

Comment (zelima) [inline tests/logic/test_logic.py]: yes

**Pull request merged**